R/read_csv.R

```diff
@@ -238,32 +238,72 @@ read_sample_csv <- function(files,
             }
           }
+        } else {
+          warmup_draws <- NULL
+          warmup_sampler_diagnostics_draws <- NULL
+          if (length(variables) > 0) {
+            new_post_warmup_draws <- posterior::as_draws_array(draws[, variables])
+            if (is.null(post_warmup_draws)) {
+              post_warmup_draws <- new_post_warmup_draws
+            } else {
+              post_warmup_draws <- posterior::bind_draws(post_warmup_draws,
+                                                        new_post_warmup_draws,
+                                                        along="chain")
+            }
+          }
+          if (sampling_info$algorithm != "fixed_param") {
+            if (length(sampler_diagnostics) > 0) {
+              new_post_warmup_sampler_diagnostics_draws <- posterior::as_draws_array(draws[, sampler_diagnostics])
+              if (is.null(post_warmup_sampler_diagnostics_draws)) {
+                post_warmup_sampler_diagnostics_draws <- new_post_warmup_sampler_diagnostics_draws
+              } else {
+                post_warmup_sampler_diagnostics_draws <- posterior::bind_draws(post_warmup_sampler_diagnostics_draws,
+                                                                              new_post_warmup_sampler_diagnostics_draws,
+                                                                              along="chain")
+              }
+            }
+          }
         }
+      } else if (sampling_info$method == "generate_quantities") {
+          generated_quantities <- posterior::bind_draws(generated_quantities,
+                                                        posterior::as_draws_array(draws),#[, variables]),
+                                                        along="chain")          
       }
+      
     }
   }
   repaired_model_params <- repair_variable_names(variables)
-  if (!is.null(warmup_draws)) {
-    posterior::variables(warmup_draws) <- repaired_model_params
-  }
-  if (!is.null(post_warmup_draws)) {
-    posterior::variables(post_warmup_draws) <- repaired_model_params
-  }
-  if (length(not_matching) > 0) {
-    not_matching_list <- paste(unique(not_matching), collapse = ", ")
-    warning("The supplied csv files do not match in the following arguments: ", not_matching_list, "!")
-  }
   sampling_info$model_params <- repair_variable_names(sampling_info$model_params)
-  sampling_info$inv_metric <- NULL
-  list(
-    sampling_info = sampling_info,
-    inv_metric = inv_metric,
-    step_size = step_size,
-    warmup_draws = warmup_draws,
-    post_warmup_draws = post_warmup_draws,
-    warmup_sampler_diagnostics = warmup_sampler_diagnostics_draws,
-    post_warmup_sampler_diagnostics = post_warmup_sampler_diagnostics_draws
-  )
+  if (sampling_info$method == "generate_quantities") {    
+    if (!is.null(generated_quantities)) {
+      posterior::variables(generated_quantities) <- repaired_model_params
+    }
+    list(
+      info = sampling_info,
```

Comment:
I dont know how to name this list element. For `sample` we named it sampling info. This question is also relevant for the other PR I am preparing to fix #199. 

Do we want to have the same name for all methods for this list element that holds all the info (args + other info needed to replicate this). Or do we go with sampling_info, gq_info, optimize_info, etc.

Reply:
My initial thought is that we should just use the same name for all of them. What do you prefer?

Reply:
Yes, I would also prefer having the same name for everything. Not sure on the name:
- args is bad because it's not just arguments
- info sounds a bit generic but might work

Other ideas:
- fit_info
- fit_settings or just setting, 
- fit_metadata or just metadata,
- rstan uses sim I think?

I dont really have a preference among them.

Reply:
Hmm, maybe metadata? I'm not sure either. For now I guess just pick whatever you want and we can do a find-and-replace before merging if we want to change it.

Reply:
Will do. Thanks!

Reply:
Adding a reminder here to eventually use this new metadata (or whatever we call it) method in `print()` as mentioned in https://github.com/stan-dev/cmdstanr/pull/221#discussion_r445130357

Reply:
Thanks. Will do it in #222  Sorry for this confusion. I just figured it made more sense.

Reply:
yeah, definitely makes more sense!